executor/builder.go

```diff
@@ -447,9 +447,9 @@ func (b *executorBuilder) buildSemiJoin(v *plan.PhysicalHashSemiJoin) Executor {
 	}
 	e := &HashSemiJoinExec{
 		schema:       v.GetSchema(),
-		otherFilter:  expression.ComposeCNFCondition(v.OtherConditions),
-		bigFilter:    expression.ComposeCNFCondition(v.LeftConditions),
-		smallFilter:  expression.ComposeCNFCondition(v.RightConditions),
+		otherFilter:  expression.ComposeConditionWithBinaryOp(v.OtherConditions, ast.AndAnd),
```

Comment:
I think ComposeCNF / DNF is better, you can call composeConditionWithBinaryOp inside expression package.